src/foam/nanos/auth/AuthenticatedUserDAO.java

```diff
@@ -0,0 +1,60 @@
+package foam.nanos.auth;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import foam.mlang.MLang;
+import foam.mlang.sink.Count;
+
+import java.security.NoSuchAlgorithmException;
+
+//TODO: Throw exception for print statements when they are ready
+public class AuthenticatedUserDAO
+  extends ProxyDAO
+{
+  public AuthenticatedUserDAO(DAO delegate) {
+    setDelegate(delegate);
+  }
+
+  @Override
+  public FObject put_(X x, FObject fObject) {
+    if ( fObject instanceof User ) {
```

Comment:
If isn't necessary, just cast in DAO's assuming correct type. Will throw ClassCastExcpetion if it isn't, which is correct behaviour.